src/mgr.sol

```diff
@@ -186,128 +205,131 @@ contract TinlakeManager {
         z = x > y ? y : x;
     }
 
+
+    // moves the rwaToken into the vault
+    // requires that mgr contract holds the rwaToken
+    function lock(uint256 wad) public auth {
+        rwaToken.approve(address(urn), uint256(wad));
+        urn.lock(wad);
+    }
+
     // --- Vault Operation---
     // join & exit move the gem directly into/from the urn
-    function join(uint256 wad) public operatorOnly {
+    function join(uint256 wad) public auth {
         require(safe && live, "TinlakeManager/bad-state");
         require(int256(wad) >= 0, "TinlakeManager/overflow");
         gem.transferFrom(msg.sender, address(this), wad);
-        vat.slip(ilk, address(this), int256(wad));
-        vat.frob(ilk, address(this), address(this), address(this), int256(wad), 0);
         emit Join(wad);
     }
 
-    function exit(uint256 wad) public operatorOnly {
+    function exit(uint256 wad) public auth {
         require(safe && live, "TinlakeManager/bad-state");
         require(wad <= 2 ** 255, "TinlakeManager/overflow");
-        vat.frob(ilk, address(this), address(this), address(this), -int256(wad), 0);
-        vat.slip(ilk, address(this), -int256(wad));
         gem.transfer(msg.sender, wad);
         emit Exit(wad);
     }
 
     // draw & wipe call daiJoin.exit/join immediately
-    function draw(uint256 wad) public operatorOnly {
+    function draw(uint256 wad) public auth {
         require(safe && live, "TinlakeManager/bad-state");
-        (, uint256 rate, , , ) = vat.ilks(ilk);
-        uint256 dart = divup(mul(RAY, wad), rate);
-        require(int256(dart) >= 0, "TinlakeManager/overflow");
-        vat.frob(ilk, address(this), address(this), address(this), 0, int256(dart));
-        daiJoin.exit(msg.sender, wad);
+        urn.draw(wad);
+        dai.transferFrom(address(this), msg.sender, wad);
         emit Draw(wad);
     }
 
-    function wipe(uint256 wad) public operatorOnly {
+    function wipe(uint256 wad) public {
         require(safe && live, "TinlakeManager/bad-state");
-        dai.transferFrom(msg.sender, address(this), wad);
-        daiJoin.join(address(this), wad);
-        (,uint256 rate, , , ) = vat.ilks(ilk);
-        uint256 dart = mul(RAY, wad) / rate;
-        require(dart <= 2 ** 255, "TinlakeManager/overflow");
-        vat.frob(ilk, address(this), address(this), address(this), 0, -int256(dart));
+        dai.transferFrom(msg.sender, address(urn), wad);
+        urn.wipe(wad);
         emit Wipe(wad);
     }
 
-    // --- Administration ---
-    function setOperator(address newOperator) external operatorOnly  {
-        operator = newOperator;
-        emit SetOperator(newOperator);
+    function free(uint256 wad) public auth {
+        urn.quit();
```

Comment:
In that case we would actually deal with it using `migrate` since we're not quite sure what we should do with this DAI after ES anyway. I think this is theoretically not a case that should happen with our use of MIP21 anyway - we wouldn't have any DAI accrue in the Urn.

Never the less, I renamed this method to `quit` to make it a bit more intuitive and actually also implemented a `free` that takes out the RWAToken.